### ISSUE TITLE
Images: Add support for previewing and adjusting the different croppings that will happen to an image

### ISSUE THREAD

**Issue by NayaMakarova** [benwixen/tripfinger-ios#208]
Horizontal images are cut / incomplete in Swiper, while when you go to description with this image - it's normal. F.ex. monk on the picture doesn't have half of the face in Swiper. Poor monk..you should help him.
![image-05-03-16-10-45](https://cloud.githubusercontent.com/assets/16025697/13546453/92d7ef26-e2bf-11e5-8ff4-11a78d4d797f.png)

**Comment by paandahl** [comment]
Yes, this is actually expected. If a picture is too big to fit, it's possible to either scale it, or crop it. Here we are using cropping. This can be adjusted - but for most pictures, cropping is fine.

Facebook f.ex. does the same. Look at this picture:

<img width="331" alt="screen shot 2016-03-05 at 15 51 08" src="https://cloud.githubusercontent.com/assets/44539/13546477/282982cc-e2ea-11e5-94a5-6723284e871f.png">

<img width="649" alt="screen shot 2016-03-05 at 15 51 17" src="https://cloud.githubusercontent.com/assets/44539/13546481/2e2ec420-e2ea-11e5-87a5-652192565449.png">

**Comment by paandahl** [comment]
Unless you want to discuss it further, you can close this issue.

**Comment by NayaMakarova** [comment]
Yes, I understand. For some pictures it's fine, but for this monk (and some others) it's not fine at all. So I shouldn't upload horizontal images?

Sent from Mail.Ru app for iOS

Saturday 5 March 2016 10:52 +0300 from notifications@github.com  notifications@github.com:

**Comment by paandahl** [comment]
It's fine :p

But it's not perfect. I will change this issue, to create a feature request. The feature will be, the ability to preview how an image will be cropped, and adjust it, from the admin.

This is low priority, and will probably not happen in the near future. In the meantime, there is a workaround:

As long as the images have the same aspect ratio as we display, they will not be cropped. So if you want to optimize it for the swiper, you can crop it manually before you upload it. It will then be cropped when it is displayed horizontally (the cropping keeps the centre of the image). The aspect ratios are documented here (I think they are up to date):

https://github.com/prebenl/tripfinger-ios/wiki/Photo-details

**Comment by paandahl** [comment]
So, if you really, really want f.ex. a head from being cropped in any of the views, you have to make sure it's not at the edges of the picture.

**Comment by NayaMakarova** [comment]
Of course I want. Would you like your head to be cropped? Ok, I understood.

Sent from Mail.Ru app for iOS

Saturday 5 March 2016 11:06 +0300 from notifications@github.com  notifications@github.com:
